### PR TITLE
Generate lazy icon properties

### LINKED ISSUE #63: [Enhancement] Lazy properties in code output
**Description**
Instead of generating a `custom` singleton pattern for every icon, we could use the `built-in` lazy functionality.
For example:

``` kotlin
val ArrowLeft: ImageVector
    get() {
        if (_ArrowLeft != null) {
            return _ArrowLeft!!
        }
        _ArrowLeft = ImageVector.Builder(....).build()

        return _ArrowLeft!!
    }

private var _ArrowLeft: ImageVector? = null
```

**becomes**

``` kotlin
val ArrowLeft: ImageVector by lazy {
     ImageVector.Builder(....).build()
}
```

A minor thing to be sure but it does:
1. Reduce indentation by one level
2. Make the singleton pattern clearer
3. Reduce the lines of code by `8 / per file` which across something like `300` icons becomes significant

P.S. Love the plugin!

### PR DESCRIPTION
**Pull request by Goooler**
- Closes #63.
- Depends on #64.

### CODE REVIEW

**Comment by egorikftp** [comment]
Wow, so quick 🚀 
Will check soon

**Comment by egorikftp** [comment]
@Goooler Lazy delegate should be as an additional option for export, not an replacement for current 🙂

**Comment by Goooler** [comment]
I'm not sure why Jetpack team used backing properties before, if there are no necessary reasons, we can fully migrate to lazy. Otherwise, we have to add an extra toggle for this, maybe with extra exporting steps?

**Comment by egorikftp** [comment]
For my project I prefer backing property approach due to it uses less overhead. Someone prefer using lazy block, due to code more compact. (not remember any existing solutions which provides lazy generation out of the box)

I think plugin support both options 🙂
You can simply create basic generation logic, I will integrate it into plugin as additional setting

File: components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorFileSpec.kt [hunk 1]
```diff
@@ -99,41 +88,30 @@ internal class ImageVectorFileSpec(private val config: ImageVectorSpecConfig) {
...
-            addStatement("return %N!!", backingProperty)
         }
+
+        delegate(
+            CodeBlock.builder()
+                .beginControlFlow("lazy(%T.NONE)", ClassNames.LazyThreadSafetyMode)
+                .add(codeBlock)
+                .endControlFlow()
+                .build(),
+        )
```
Comment (Goooler) [inline components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorFileSpec.kt:114]: All here, you can copy and paste anytime. I'll leave the PR here until I have more time to take or finish by you.

**Comment by Goooler** [comment]
Closed due to #70.

**Pull request closed without merging**